### ISSUE TITLE
[BUG] multi line footnotes catastrophe linter moves only first line !!!

### ISSUE THREAD

**Issue by Parlazo** [platers/obsidian-linter#83]
if i do text 

```
lora ipsala [^note] dolores.

[^note]: This is not right

    - because the footnote takes more than one paragraph.
    - and linter does only move the first line.

    but not the rest of it

    it will mess up every multiline note instantly

this is not cool
```

it will destroy all all footnotes

**Comment by chrisgrieser** [comment]
I looked into this and it actually seems that Obsidian does not recognize multi-line-notes as such:
<img width="673" alt="Screenshot 2021-09-27 00 47 27" src="https://user-images.githubusercontent.com/73286100/134826729-46295f37-3c06-4361-94ea-87d2b6baaae9.png">

I am not totally sure whether multi-line-footnotes are part of the markdown specification. But even if they are and this is a bug on Obsidian's side, I am unsure whether the Linter plugin should implement this when Obsidian clearly does not seem to recognize multi-line-footnotes.

**Comment by Parlazo** [comment]
this is what obsidian tells me in the obsidian help 

under `Format your notes`

``````
### Footnotes

```md
Here's a simple footnote,[^1] and here's a longer one.[^bignote]

[^1]: meaningful!

[^bignote]: Here's one with multiple paragraphs and code.

    Indent paragraphs to include them in the footnote.

    `{ my code }`

    Add as many paragraphs as you like.
```

Here's a simple footnote,[^1] and here's a longer one.[^bignote]

[^1]: meaningful!

[^bignote]: Here's one with multiple paragraphs and code.

    Indent paragraphs to include them in the footnote.

    `{ my code }`

    Add as many paragraphs as you like.
	
```md
You can also use inline footnotes. ^[notice that the carat goes outside of the brackets on this one.]
```

You can also use inline footnotes. ^[notice that the carat goes outside of the brackets on this one.]
``````

and it seems to work, and if linter is uninstalled it all keeps together where it is
with linter installed and even deactivated it always moves the first line and the block below does not move and stays as a meaningles fragment at the place

**Comment by chrisgrieser** [comment]
Hmmm. I see. Still, Obsidian does display multi-line-footnotes wrong in the editor.

Nevertheless, implementing it with regexes could be quite finicky and is likely prone to bugs. I think this  will be best addressed with the markdown parser platers is looking into? (#76)

**Comment by Parlazo** [comment]
``````
Some Text

Some Footnote will be here [^Footnote] inline the text.

[^Footnote]: Below

    this is a footnote
	and it contains a list
	
	- this is a item
	
	- this is cool
	
	```
	bsidian does display this
	```
	
	pretty nice

this is text below the footnote and multiple footnotes can follow
``````
![LinterFootnote](https://user-images.githubusercontent.com/66982240/135719821-0b23bbd9-a155-4f77-aa5a-e2d66c87ebc0.png)

This is a multiline footnote in obsidian and i think the display is not wrong. When clicking on it Obsidian does jump to it and highlight it for an second as shown. The footnote contains text a list and code, below the footnote the regular text goes on.

I think linter should move such blocks as a block to the bottom and never just the first line because it will disassemble all footnotes instantly and break them and a user has then to uninstall (not only deactivate) linter and then sort all the initial lines of every footnote block manually back to fix them.

**Comment by Parlazo** [comment]
![LinterFootnote](https://user-images.githubusercontent.com/66982240/135720632-3035a750-2ce4-45b7-a98d-0c6e1b19e9ef.png)
Here you see the same footnote with the `better footnote` plugin as a popup instead of a seperate footnote section that is seperated from the main text atthe bottom of the note.
https://github.com/aidenlx/better-fn

how ever if you use such multiline footnotes while linter does only move the first anchor-line to the bottom while the rest stays disconnected in the note where it is, it will messup all footnotes and you have to reasign all anchor-lines to the right footnotes, and as soon you save the note you have to do it again until you uninstall linter completely.
this is may not so problematic if you use 3 footnotes. But if you use 100 footnotes it will not get boring when reassigning all anchor lines to the correct footnote.

**Comment by Parlazo** [comment]
``````
Some Text

Some Footnote will be here [^Footnote] inline the text.

    this is a footnote
	and it contains a list
	
	- this is a item
	
	- this is cool
	
	```
	bsidian does display this
	```
	
	pretty nice

this is text below the footnote and multiple footnotes can follow

[^Footnote]: Below
``````
this is what linter does when moving all footnotes to the bottom of the note
you see again the same footnote
but after linters moving action the anchor of the footnote is at the bottom while the rest stays where it is, and this will break the whole footnote block as one block and reduce it to the anchor line while the rest of the block will look like code out of context.

**Comment by Parlazo** [comment]
My obsidian version = 0.12.15
My Linter version = 1.2.2
Both on windows

**Comment by chrisgrieser** [comment]
when I said that multi-line-footnotes do not work properly, I was referring to the edit mode.
![Screenshot 2021-10-02 16 48 41](https://user-images.githubusercontent.com/73286100/135721730-12bb8a96-91ec-4e14-812f-3673e5bf71d6.png)

Otherwise, I have already understood your issue the first time and already said that it will be addressed, pending on issue #76 which is needed for that.

Also as a side note, please use the edit-button when you want to add something after submitting a comment. Reason is that anyone subscribed to this issue (or all issues of this repository) will get a separate notification for each comment you make, which can be a bit annoying.
